samples/src/Microsoft.Azure.IIoT.App/src/Shared/WorkbookLink.razor

```diff
@@ -0,0 +1,56 @@
+﻿@*------------------------------------------------------------
+  Copyright (c) Microsoft Corporation.  All rights reserved.
+  Licensed under the MIT License (MIT). See License.txt in the repo root for license information.
+ ------------------------------------------------------------*@
+
+@using Microsoft.Azure.IIoT.App.Runtime
+@using Microsoft.AspNetCore.Components
+@using System
+
+@inject NavigationManager NavigationManager
+@inject Config Configuration
+
+<AuthorizeView>
+    <Authorized>
+        @if(!String.IsNullOrEmpty(_workbookLink))
+        {
+            <a href="@_workbookLink" target="_blank">Go to IIoT Workbook</a>
+        }
+    </Authorized>
+    <NotAuthorized>
+    </NotAuthorized>
+</AuthorizeView>
+
+@code {
+    private string _workbookLink { get; set; } = null;
+
+
+    /// <summary>
+    /// OnInitialized
+    /// </summary>
+    protected override void OnInitialized()
+    {
+        CreateWorkbookLink();
+    }
+
+    /// <summary>
+    /// Create workbook link
+    /// </summary>
+    private void CreateWorkbookLink()
+    {
+        if (!String.IsNullOrEmpty(Configuration.WorkbookId))
+        {
+            var index = Configuration.AuthAudience.LastIndexOf('/');
+            if (index > 0)
+            {
+                _workbookLink = "https://portal.azure.com/#@" +
```

Comment:
@dacolgit, this whole workbook link generation seems to be super hacky and error prone. And it seems that you are mostly generating resource ID of the workbook (`/subscriptions/<subscription_id>/resourceGroups/<rg_name>/providers/Microsoft.OperationalInsights/workspaces/<name>`). Wouldn't it be easier and more correct if you get that value from output of `azuredeploy.workbook.json` template deployment and store it as whole in a PCS variable ?

Reply:
changed

Reply:
Just a note, `azuredeploy.workbook.json` actually outputs a `workbookId` (which can be renamed to `workbookUri` to differentiate it from input `workbookId`) which would most probably contain whole URI for the workbook in `/subscriptions/<subscription_id>/resourceGroups/<resource_group_name>/providers/Microsoft.OperationalInsights/workspaces/<workspaces_name>`. That can be saved to KeyVault and used as a PCS var.

Reply:
from docs (https://docs.microsoft.com/en-us/azure/azure-monitor/platform/workbooks-automate) the output is the workbookId not the Uri